stock_reserve/models/product.py

```diff
@@ -46,6 +46,23 @@ def _compute_reservation_count(self):
             reservations = self.env["stock.reservation"].search(domain)
             product.reservation_count = sum(reservations.mapped("product_qty"))
 
+    def _get_domain_locations_new(
+        self, location_ids, company_id=False, compute_child=True
+    ):
+        """Add location reservation for avoiding reservations to be substracted on
+        the forecasted quantity.
+        """
+        company = self.company_id or self.env.company
```

Comment:
This is not valid, as products by default don't have company, so this is always going to take the main company, which is not what is needed, specially on things coming from OdooBot. Argument `company_id` is not filled? If any doubt about the company passed, please go back to what I asked to put this on custom.

Reply:
Ok, taking into account what you indicate, we add this to custom (subtract always) and leave this PR to implement option 1 (more difficult to implement)

Reply:
But it won't be valid if the company is not perfectly guessed.

Reply:
With option 1 yes: Define for each reservation whether to subtract or not (auto-defining the value of a new field according to the company configuration), but the filter to be performed would be more complex than just by location.